File: VB/WindowsApplication1/Custom DataNavigator/MyNavigatorButtonsViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	Public Class MyNavigatorButtonsViewInfo
		Inherits NavigatorButtonsViewInfo
		Public Sub New(ByVal buttons As NavigatorButtonsBase)
			MyBase.New(buttons)

		End Sub

		Protected Overrides Function CreateTextViewInfo() As NavigatorTextViewInfo
			Return New MyNavigatorTextViewInfo(Me)
		End Function

	End Class
End Namespace

File: CS/WindowsApplication1/Custom DataNavigator/MyNavigatorButtonsViewInfo.cs
using System;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    public class MyNavigatorButtonsViewInfo : NavigatorButtonsViewInfo
    {
        public MyNavigatorButtonsViewInfo(NavigatorButtonsBase buttons)
            : base(buttons)
        {

        }

        protected override NavigatorTextViewInfo CreateTextViewInfo()
        {
            return new MyNavigatorTextViewInfo(this);
        }

    }
}
